include/drivers/gpio.h

```diff
@@ -30,14 +30,59 @@ extern "C" {
  * @brief GPIO Driver APIs
  * @defgroup gpio_interface GPIO Driver APIs
  * @ingroup io_interfaces
+ *
+ * @{
+ * @name GPIO Atomicity and Interrupt-Related Behavior
+ *
  * @{
+ * All GPIO peripheral API operations satisfy the following promises:
+ *
+ * * All functions shall be thread-safe and interrupt-safe.
+ * * All functions may be invoked from interrupt service routines and
+ *   callbacks.  Functions that cannot fulfill their behavioral
+ *   requirements in that context may return `-EWOULDBLOCK`.
```

Comment:
It should be rather "must" or "should" return `-EWOULDBLOCK`. This shouldn't be optional.

Reply:
done